src/sentry/static/sentry/app/views/performance/realUserMonitoring/vitalCard.tsx

```diff
@@ -0,0 +1,396 @@
+        x: summaryBucket - 0.5,
+        y: Math.max(...chartData.map(data => data.count)) || 1,
+      },
+      this.state.refDataRect!,
+      this.state.refPixelRect!
+    );
+
+    series.markLine = MarkLine({
+      data: [[thresholdPixelBottom, thresholdPixelTop] as any],
+      label: {
+        show: false,
+      },
+      lineStyle: {
+        color: theme.gray700,
+        type: 'solid',
+      },
+      silent: true,
+    });
+  }
+
+  drawFailRegion(series) {
+    const {chartData, vital} = this.props;
+    const {failureThreshold} = vital;
+    if (this.state.refDataRect === null || this.state.refPixelRect === null) {
+      return;
+    }
+
+    const failureBucket = findNearestBucketIndex(
+      chartData,
+      this.bucketWidth(),
+      failureThreshold
+    );
+    if (failureBucket === null) {
+      return;
+    }
+
+    // since we found the failure bucket, the failure threshold is
+    // visible on the graph, so let's draw the fail region
+    const failurePixel = asPixel(
+      {
+        // subtract 0.5 from the x here to ensure that the boundary of
+        // the failure region lies between buckets
+        x: failureBucket - 0.5,
+        y: 0,
+      },
+      this.state.refDataRect!,
+      this.state.refPixelRect!
+    );
+
+    series.markArea = MarkArea({
+      data: [
+        [
+          {x: failurePixel.x, yAxis: 0},
+          {x: 'max', y: 'max'},
+        ] as any,
+      ],
+      itemStyle: {
+        color: 'transparent',
+        borderColor: theme.red400,
+        borderWidth: 1.5,
+        borderType: 'dashed',
+      },
+      silent: true,
+    });
+
+    const topRightPixel = asPixel(
+      {
+        // subtract 0.5 to get on the right side of the right most bar
+        x: chartData.length - 0.5,
+        y: Math.max(...chartData.map(data => data.count)) || 1,
+      },
+      this.state.refDataRect!,
+      this.state.refPixelRect!
+    );
+
+    series.markPoint = MarkPoint({
+      data: [{x: topRightPixel.x - 16, y: topRightPixel.y + 16}] as any,
+      itemStyle: {color: theme.red400},
+      silent: true,
+      symbol: `path://${FIRE_SVG}`,
+      symbolKeepAspect: true,
+      symbolSize: [14, 16],
+    });
+  }
+
+  render() {
+    return (
+      <Card>
+        {this.renderSummary()}
+        {this.renderHistogram()}
+      </Card>
+    );
+  }
+}
+
+type IndicatorProps = {
+  color: string;
+};
+
+const Indicator = styled('div')<IndicatorProps>`
+  position: absolute;
+  left: 0px;
+  margin-top: ${space(0.5)};
+  width: 6px;
+  height: 18px;
+  border-radius: 0 3px 3px 0;
+  background-color: ${p => p.color};
+`;
+
+type TagProps = {
+  color: string;
+};
+
+const StyledTag = styled(Tag)<TagProps>`
+  position: absolute;
+  right: ${space(3)};
+  background-color: ${p => p.color};
+  color: ${p => p.theme.white};
+  text-transform: uppercase;
+  font-weight: 500;
+`;
+
+function formatDuration(duration: number) {
```

Comment:
We've done sub-second duration as a special format a few times now, maybe not in the PR but perhaps we should move it into a util 🤔

Reply:
@k-fish we're specializing it here since some histogram bins could be merged accidentally by the charting library. 

For example: `1.112 sec` and `1.114 sec` could end up into the same bin on echarts.

Reply:
We can backlog this, this has come up a few times now and may be useful as a higher level utility.